src/engine/TransitivePathBinSearch.h

```diff
@@ -0,0 +1,216 @@
+// Copyright 2019, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Florian Kramer (florian.kramer@neptun.uni-freiburg.de)
+
+#pragma once
+
+#include <iterator>
+#include <memory>
+
+#include "TransitivePathBase.h"
+#include "engine/Operation.h"
+#include "engine/QueryExecutionTree.h"
+#include "engine/idTable/IdTable.h"
+
+struct BinSearchMap {
+  std::span<const Id> startIds_;
+  std::span<const Id> targetIds_;
+
+  std::span<const Id> successors(const Id node) const {
```

Comment:
Yes, this can also be an `auto` type.

But your one-liner solution does not work, because we need to return the elements from the `targetIds_`, not the `startIds_`.